Update NXP LCDIF driver

mmahadevan108 opened the pull request:
1. Rename device data to dev_data to fix variable name clash
2. Use Device Tree properties to setup the display
3. Delete unused Kconfigs

mmahadevan108 committed:
drivers: display: Update the MCUX ELCDIF driver

1. Rename device data to dev_data to fix variable
   name clash
2. Use Device Tree properties to setup the display
3. Delete unused Kconfigs

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
boards: Update MXRT display settings

Add display properties for display RK043FN02H

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

gmarull commented on dts/bindings/display/fsl,imx6sx-lcdif.yaml:
```diff
@@ -14,5 +14,94 @@ properties:
+        0 RGB_888.
+        1 BGR_565.
+
+    data-buswidth:
+      type: int
+      enum:
+        - 0
+        - 1
+        - 2
+        - 3
+      description:
+        LCD data bus
+        0 16-bit data bus mode.
+        1 8-bit data bus mode.
+        2 18-bit data bus mode.
+        3 24-bit data bus mode.
```
```suggestion
    data-buswidth:
      type: string
      enum:
        - "16-bit"
        - "8-bit"
        - "18-bit"
        - "24-bit"
      description:
        LCD data bus width.
```

and use `DT_ENUM_IDX()` helper

mmahadevan108 replied:
Thanks for the suggestion, I have made the change.

gmarull commented on dts/bindings/display/fsl,imx6sx-lcdif.yaml:
```diff
@@ -14,5 +14,94 @@ properties:
+        1111 VSYNC active high, HSYNC active high, Drive data on rising edge, Data enable active high.
+
+    pixel-format:
+      type: int
+      enum:
+        - 0
+        - 1
+      description:
+        Display pixel format
+        0 RGB_888.
+        1 BGR_565.
```
```suggestion
      type: int
      enum:
        - "rgb-888"
        - "bgr-565"
      description:
        Display pixel format
```

and use DT_ENUM_IDX()

mmahadevan108 replied:
Thanks for the suggestion, I have made the change.

gmarull commented on drivers/display/display_mcux_elcdif.c:
```diff
@@ -47,44 +51,44 @@ static int mcux_elcdif_write(const struct device *dev, const uint16_t x,
 {
 	const struct mcux_elcdif_config *config = dev->config;
-	struct mcux_elcdif_data *data = dev->data;
+	struct mcux_elcdif_data *dev_data = dev->data;
```
PR mentions there is a clash, where?

mmahadevan108 replied:
Similar to the change in PR https://github.com/zephyrproject-rtos/zephyr/pull/42077/files

gmarull reviewed (commented):
A few non-blocking suggestions for the DT part

hakehuang commented:
@mmahadevan108  CI testing PASS and manual check on rt1050  for display PASS

danieldegrasse commented on dts/bindings/display/fsl,imx6sx-lcdif.yaml:
```diff
@@ -1,11 +1,11 @@
 
 description: NXP i.MX eLCDIF (Enhanced LCD Interface) controller
 
 compatible: "fsl,imx6sx-lcdif"
```
Should we change this compatible string to reflect the fact that the `lcdif` IP block is present on devices besides the iMX6 line?

mmahadevan108 committed:
dts: Rename NXP LCDIF dts node

Rename fsl,imx6sx-lcdif to nxp,imx-elcdif as it
used by imxrt and imx platforms.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
dts: Update NXP LDCIF to add display properties

Add properties to specify panel properties

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mbolivar-nordic commented on dts/bindings/display/nxp,imx-elcdif.yaml:
```diff
@@ -14,5 +14,87 @@ properties:
+      type: int
       required: true
+      description: HSYNC pulse width
+    hfp:
+      type: int
+      required: true
+      description: Horizontal front porch
+
+    hbp:
+      type: int
+      required: true
+      description: Horizontal back porch
+
+    vsync:
+      type: int
+      required: true
+      description: VSYNC pulse width
+
+    vfp:
+      type: int
+      required: true
+      description: Vertical front porch
+
+    vbp:
+      type: int
+      required: true
+      description: Vertical back porch
```
For all of these: units?

mmahadevan108 replied:
Thank you, I have added the units in the description

mbolivar-nordic commented on drivers/display/display_mcux_elcdif.c:
```diff
@@ -227,47 +257,43 @@ static const struct display_driver_api mcux_elcdif_api = {
+			.vbp = DT_INST_PROP(id, vbp),				\
+			.polarityFlags = DT_INST_PROP(id, polarity),		\
+			.dataBus = LCDIF_CTRL_LCD_DATABUS_WIDTH(		\
+					DT_INST_ENUM_IDX(id, data_buswidth)),	\
+		},								\
+		.pixel_format = DT_INST_ENUM_IDX(id, pixel_format),		\
```
Thanks for adding the units. Looking at the uses in the driver, these values are being read unconditionally, so it also looks like the binding should make them `required: true`.

mmahadevan108 replied:
Thank you for your comment, I have update the dts file.

mbolivar-nordic replied:
@mmahadevan108 I don't understand your response, sorry. My request was related to the binding, not the DTS file:

> the binding should make them required: true.

I don't see this change

mmahadevan108 replied:
I have added a default value instead, is'nt that similar to making them required?

mbolivar-nordic replied:
Oh, sorry, I missed that. But now this is breaking the rules about using default without a justification in the description, so that needs to be fixed, please.

mmahadevan108 replied:
Thanks, updated.

hakehuang reviewed (dismissed):
Test PASS on my rt1050 boards, and build pass on all rt10xx seriese platform with LCD interfaces

mmahadevan108 commented:
@danieldegrasse @dleach02 @mbolivar-nordic Can you please re-review.

mbolivar-nordic reviewed (dismissed):
DT portions look fine to me.

@mmahadevan108 in the future, though, can you please structure your commits to be bisectable? The current PR has the bindings patch *after* the patches that use the new properties in the binding in drivers, so they would fail to compile if you try to build anything using this driver before the last commit.

To avoid this issue in the future, please commit the bindings changes first, before the patches that use them.

mmahadevan108 commented:
> DT portions look fine to me.
> 
> @mmahadevan108 in the future, though, can you please structure your commits to be bisectable? The current PR has the bindings patch _after_ the patches that use the new properties in the binding in drivers, so they would fail to compile if you try to build anything using this driver before the last commit.
> 
> To avoid this issue in the future, please commit the bindings changes first, before the patches that use them.

Thanks @mbolivar-nordic. I have fixed the commit order to address the bisectable issue.
